File: src/Test/TCOFFICheatcode.erl
-module(test_tCOFFICheatcode@foreign).
-export([runTCOFFICheatcode/1]).
runTCOFFICheatcode(N) -> sum(N, 0).
sum(0, Acc) -> Acc;
sum(N, Acc) -> sum(N - 1, Acc + 1).

File: src/Test/TCOFFICheatcode.java
public static final java.util.function.Function<Object, Object> runTCOFFICheatcode = Bench.nativeBenchmark(input -> {
    int n = ((Number) input).intValue();
    int acc = 0;
    while (n > 0) {
        acc += n % 3;
        n--;
    }
    return acc;
});
